File: ArtMuseum/python/makers.py
from configs import MAKERS
from req_lib import getJSON
import json

'''
Prints information related to makers in
the Princeton University Art Museum's
collection.
'''

if __name__ == "__main__":
    maker_name = "Pablo Picasso"
    with open("makers.json", "r") as makefile:
        data = json.loads(makefile.read())
    
    for m in data:
        if m["displayname"] == maker_name:
            maker_id = m["ConstituentID"]
            break

    endpoint = MAKERS + str(maker_id)
    makers = getJSON(endpoint)
    print(makers)

File: ArtMuseum/java/Makers.java
import java.io.FileInputStream;
import java.util.HashMap;

import javax.json.Json;
import javax.json.JsonArray;
import javax.json.JsonObject;
import javax.json.JsonReader;

/*
Prints information related to makers in
the Princeton University Art Museum's
collection.
*/

public class Makers {
    public static void main(String[] args) {
        try {
            ReqLib reqLib = new ReqLib();
            String maker_name = "Pablo Picasso";

            FileInputStream stream = new FileInputStream("makers.json");
            JsonReader jsonReader = Json.createReader(stream);
            JsonArray data = jsonReader.readArray();
            jsonReader.close();
            stream.close();

            int maker_id = 0;
            for (int i = 0; i < data.size(); i++) {
                JsonObject m = data.getJsonObject(i);
                if (m.getJsonString("displayname").getString().equals(maker_name)) {
                    maker_id = m.getJsonNumber("ConstituentID").intValue();
                    break;
                }
            }

            String endpoint = reqLib.configs.makers + maker_id;
            HashMap<String, Object> params = new HashMap<String, Object>();

            String makers = reqLib.makeRequest(endpoint, params);
            System.out.println(makers);
        } catch (Exception e) {
            System.err.println(e);
        }
    }
}